scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,217 @@
+            short flagsEnd = (Short) flagsField.get(end[0]);
+            flagsStart |= FLAG_RESOLVED;
+            flagsEnd |= FLAG_RESOLVED;
+
+            flagsField.set(start[0], flagsStart);
+            bytecodeOffsetField.set(start[0], scopeStart);
+
+            flagsField.set(end[0], flagsEnd);
+            bytecodeOffsetField.set(end[0], scopeStart + scopeLength);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
     }
 
+    /**
+     * Returns the bytecode offset to the start of the scope.
+     * @return The bytecode offset to the start of the scope
+     */
+    public int getScopeStart() {
+        try {
+            return start[0].getOffset();
+        } catch (IllegalStateException e) {
+            System.err.println("Labels not resolved: " + Arrays.toString(start));
+            return -1;
+        }
+    }
 
     /**
-     * Returns whether this {@link LocalLocation} equals <code>o</code>; a
-     * slightly faster variant of {@link #equals(Object)} for when the argument
-     * is statically known to be another nonnull {@link LocalLocation}.
+     * Returns the length of the scope (in bytes).
+     * @return The length of the scope (in bytes)
      */
-    public boolean equals(LocalLocation l) {
-        return index == l.index && scopeStart == l.scopeStart
-                && scopeLength == l.scopeLength &&
-                (varName==null || varName.equals(l.varName)) &&
-                varIndex==l.varIndex;
+    public int getScopeLength() {
+        // Should this be end[0] instead?
+        try {
+            return end[end.length - 1].getOffset() - getScopeStart();
+        } catch (IllegalStateException e) {
+            System.err.println("Labels not resolved: " + Arrays.toString(end));
+            return -1;
+        }
     }
 
     /**
-     * This {@link LocalLocation} equals <code>o</code> if and only if
-     * <code>o</code> is another nonnull {@link LocalLocation} and
-     * <code>this</code> and <code>o</code> have equal {@link #index},
-     * {@link #scopeStart}, and {@link #scopeLength}.
+     * Returns the local variable index.
+     * @return The local variable index
      */
+    public int getVarIndex() {
+        return index[0];
+    }
+
     @Override
     public boolean equals(Object o) {
-        return o instanceof LocalLocation
-                && equals((LocalLocation) o);
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        LocalLocation that = (LocalLocation) o;
+        return Arrays.equals(start, that.start) &&
+            Arrays.equals(end, that.end) &&
+            Arrays.equals(index, that.index);
     }
 
     @Override
     public int hashCode() {
-        if (varName==null) {
-            return Objects.hash(index, scopeStart, scopeLength);
-        } else {
-            return Objects.hash(varName, varIndex);
-        }
+        int result = Arrays.hashCode(start);
```

Comment:
This should use Objects.hash:
```
Object.hash(Arrays.hashCode(start), Arrays.hashCode(end), Arrays.hashCode(index));
```

Put the fields in an order consistent with their declarations, throughout the class.